CONTRIBUTING.md

```diff
@@ -0,0 +1,44 @@
+# Contributing
+
+## Reporting Issues
+
+If you have found what you think is a bug, please [file an issue](https://github.com/pmndrs/jotai/issues/new).
+
+## Suggesting new features
+
+If you are here to suggest a feature, first create an issue if it does not already exist. From there, we will discuss use-cases for the feature and then finally discuss how it could be implemented.
+
+## Development
+
+If you have been assigned to fix an issue or develop a new feature, please follow these steps to get started:
+
+- Fork this repository
+- Install dependencies by running `$ yarn`
+- Link `jotai` locally by running `$ yarn link`
+- Auto-build files as you edit by running `$ yarn start`
+- Implement your changes and tests to files in the `src/` directory and corresponding test files
+- To run examples, follow their individual directions. Usually this is just `$ yarn && yarn start`.
+- To run examples using your local build, link to the local `jotai` by running `$ yarn link jotai` from the example's directory
+- Git stage your required changes and commit (see below commit guidelines)
+- Submit PR for review
```

Comment:
If you think this is precise and helps developers, it's fine.

(I just usually write failing tests first, and don't follow these steps, skipping the third one.)

Reply:
Working on `jotai` today it was much more productive for me to use @dai-shi's recommended approach of writing tests and re-running them automatically via `yarn test:dev`. I think this could be a much better main approach to recommend (side effect being of encouraging TDD here since tests are already quite comprehensive)

Reply:
@israelidanny that's great to hear.

Reply:
what if we have both approaches. I think testing on a real project for fixing bugs or repos that users on github issues are reporting could be efficient. 
maybe I am wrong.

Reply:
Having both approaches is fine. I just want you to write what you are convinced with your experience. Not hypothetical one. If you don't have any concrete idea, we can drop the entire section for now.